### PR TITLE
Fix some errcheck errors

### PR DESCRIPTION
**Pull request by krishicks**
I encourage reviewers to review this on a per-commit basis for clarity.

This is mostly scoped to the structs package, except for the changes to `multierror.Append`, which were done repo-wide.

The behavior here is all unchanged except for the changes in `nomad/struts/testing.go`, which I think we would want, but maybe not.

This does not include fixing errcheck errors in `nomad/structs/csi.go`, which do result in changed behavior in production code, and which are covered in https://github.com/hashicorp/nomad/pull/9803.

### CODE REVIEW

File: nomad/structs/funcs.go [hunk 1]
```diff
@@ -18,33 +18,28 @@ import (
...
+	return mErr.Error()
 }
 
 // warningsFormatter is used to format job warnings
 func warningsFormatter(es []error) string {
-	points := make([]string, len(es))
-	for i, err := range es {
-		points[i] = fmt.Sprintf("* %s", err)
+	sb := strings.Builder{}
+	sb.WriteString(fmt.Sprintf("%d warning(s):\n", len(es)))
```
Comment (tgross) [inline nomad/structs/funcs.go:36]: `MergeMultiErrorWarnings` has a bunch of callers from different Job RPC endpoints. Isn't the command that called into the endpoint is already writing this value out? (Or even something specific to the command?) We probably shouldn't be changing the output in a static analysis refactoring PR.

File: nomad/structs/testing.go [hunk 2]
```diff
@@ -90,7 +90,10 @@ func MockNode() *Node {
 		Status:                NodeStatusReady,
 		SchedulingEligibility: NodeSchedulingEligible,
 	}
-	node.ComputeClass()
+	err := node.ComputeClass()
+	if err != nil {
+		panic(fmt.Sprintf("failed to compute node class: %v", err))
```
Comment (tgross) [inline nomad/structs/testing.go:95]: It's unfortunate that we don't have these functions in a package where they can never be called from production code. But a quick look for all callers shows we're safe.

**Review by tgross — CHANGES_REQUESTED** [review]

File: nomad/structs/funcs.go (hunk 1 above)
Comment (krishicks) [reply nomad/structs/funcs.go:36]: The associated test for this was written against the original implementation, so no change in behavior should exist.

Can you give an example where the updated implementation would return a different result than before? It looked like the only test cases were there are no errs, so we should return empty string, or there is at least one err, which means we should return the formatted warnings string.

Comment (tgross) [reply nomad/structs/funcs.go:36]: Err... no, I'm just a dope. I thought we were adding text but it's right [here](https://github.com/hashicorp/nomad/pull/9811/files#diff-958b4ba4f1d66baa7352c1ac855afee4c691da951d5c2c5ffef5e45858f1933dL46). 😊 

LGTM

**Review by tgross — APPROVED** [review]

**Pull request merged**